cli/flox-rust-sdk/src/models/manifest/composite/shallow.rs

```diff
@@ -0,0 +1,499 @@
+                (Some(h1), None) => Some(h1.clone()),
+                (None, Some(h2)) => Some(h2.clone()),
+                (None, None) => None,
+            };
+            prop_assert_eq!(merged.fish, expected);
+        }
+
+        #[test]
+        fn merges_options_section(options1 in any::<ManifestOptions>(), options2 in any::<ManifestOptions>()) {
+            let merged = ShallowMerger::merge_options(&options1, &options2).unwrap();
+            let systems = options2.systems.or(options1.systems);
+            let allow = Allows {
+                unfree: options2.allow.unfree.or(options1.allow.unfree),
+                broken: options2.allow.broken.or(options1.allow.broken),
+                licenses: if options2.allow.licenses.is_empty() { options1.allow.licenses.clone()} else { options2.allow.licenses.clone() }
+            };
+            let semver = SemverOptions { allow_pre_releases: options2.semver.allow_pre_releases.or(options1.semver.allow_pre_releases) };
+            let cuda_detection = options2.cuda_detection.or(options1.cuda_detection);
+            let expected = ManifestOptions { systems, allow, semver, cuda_detection, };
+            prop_assert_eq!(merged, expected);
+        }
+
+        #[test]
+        fn containerize_cfg_shallow_merges_user(
+            cfg_lp in any::<ManifestContainerizeConfig>(),
+            cfg_hp in any::<ManifestContainerizeConfig>(),
+        ) {
+            let merged = deep_merge_optional_containerize_config(Some(&cfg_lp), Some(&cfg_hp)).unwrap();
+            if cfg_hp.user.is_some() {
+                prop_assert_eq!(merged.user, cfg_hp.user);
+            } else {
+                prop_assert_eq!(merged.user, cfg_lp.user);
+            }
+        }
+
+        #[test]
+        fn containerize_cfg_deep_merges_ports(
+            cfg_lp in any::<ManifestContainerizeConfig>(),
+            cfg_hp in any::<ManifestContainerizeConfig>(),
+        ) {
+            let merged = deep_merge_optional_containerize_config(Some(&cfg_lp), Some(&cfg_hp)).unwrap();
+            match (cfg_lp.exposed_ports, cfg_hp.exposed_ports) {
+                (None, None) => prop_assert!(merged.exposed_ports.is_none()),
+                (Some(lp), None) => {
+                    prop_assert_eq!(merged.exposed_ports, Some(lp));
+                }
+                (None, Some(hp)) => {
+                    prop_assert_eq!(merged.exposed_ports, Some(hp));
+                }
+                (Some(lp), Some(hp)) => {
+                    prop_assert!(merged.exposed_ports.is_some());
+                    let merged_ports = merged.exposed_ports.unwrap();
+                    for port in merged_ports.iter() {
+                        if hp.contains(port) {
+                            prop_assert_eq!(merged_ports.get(port), hp.get(port));
+                        } else {
+                            prop_assert_eq!(merged_ports.get(port), lp.get(port));
+                        }
+                    }
```

Comment:
how is that different from either of these, considering these are sets.

```suggestion
                        prop_assert!(hp.contains(port) || lp.contains(port));
                    }
```

or 

```
                (Some(lp), Some(hp)) => {
                    prop_assert!(merged.exposed_ports.is_some());
                    let merged_ports = merged.exposed_ports.unwrap();
                    prop_assert_eq!(merged_ports, hp.union(lp)
                }

```

Reply:
Edit: I wrote a comment and realized it was more true for maps instead of sets, and I probably just copied and pasted a map test to create this one. The test as written is still valid but you're correct that there's a more convenient syntax.

Reply:
Fixed